File: src/models/StoryInfos.ts
import {Entity,PrimaryGeneratedColumn,Column,ManyToOne,JoinColumn} from 'typeorm';
import { Story } from './Story';

@Entity('story_infos')
export class StoryInfos {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column()
  story_id!: number;

  @Column({ type: 'varchar', nullable: true })
  about!: string;

  @Column({ type: 'varchar', nullable: true })
  description!: string;

  @ManyToOne(() => Story, (story) => story.infos)
  @JoinColumn({ name: 'story_id' })
  story!: Story;
}

File: backend/src/models/story_infos.py
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from src.db.database import Base


class StoryInfos(Base):
    __tablename__ = "story_infos"

    id = Column(Integer, primary_key=True, index=True)
    story_id = Column(Integer, ForeignKey("story.id"), nullable=False)
    about = Column(String, nullable=True)
    description = Column(String, nullable=True)

    story = relationship("Story", back_populates="infos")
